portality/dao.py

```diff
@@ -964,6 +966,25 @@ def refresh():
     return ES.indices.refresh()
 
 
+def find_indexes_by_prefix(index_prefix) -> list[str]:
+    data = ES.indices.get(f'{index_prefix}*')
```

Comment:
it might be good to enforce the separator here (`if not index_prefix.endswith('-') ...`)  because here is where a missing separator could match a whole lot more indexes than we indend. E.g. if you supply `doaj` as the index prefix you'll delete indexes doajstatic-article etc. That won't happen unless the comment in settings.py to include the separator is ignored.

Perhaps we should just ensure the setting always ends with `-` instead. Either way, not a blocker for release.

Reply:
Not recommend it.

It will fail the first time import. The first time import meaning when the the index name still calling `doaj-account` instead of `doaj-account-20111212`, which mean it will help to migrate if the environment still using old index name `doaj-account`.

Moreover, prefix search by `index_type` and  the script will show which index will be removed, it should be safe enough.